subcommands/keys/rotate_root.go

```diff
@@ -48,6 +49,10 @@ func doRotateRoot(cmd *cobra.Command, args []string) {
 	subcommands.DieNotNil(err)
 
 	if initialRotation {
+		if _, err := os.Stat(credsFile); err == nil {
+			subcommands.DieNotNil(errors.New("Destination file exists. Please make sure you aren't accidentally overwriting another factory's keys"))
+		}
```

Comment:
There is one more use case when things can go wrong.
At one point I've broken my factory by making the initial rotation into the inaccessible folder.

So, I think it makes sense to do some magic with `touch credsFile` after this to make sure the folder exists and is writable.
Currently, we rotate first, and try to write next... and if that write fails we're doomed.

I know we cannot prevent all use cases when it can break, but at least protecting against silly mistakes is worth it.

Reply:
Sorry @StealthyCoder please, disregard this.  I always forget that we write first/post next

Reply:
this branch doesn't rotate anything - its just downloading and saving *before* we get to the fun stuff. We also try and make sure you don't lose things if the rotate works in the cloud but fails to persist locally:

https://github.com/foundriesio/fioctl/blob/main/subcommands/keys/rotate_root.go#L105-L124

Reply:
Added a check to make sure directory exists, and is writable. :+1:

Reply:
@vkhoroz you are still right, because if os.Create fails because the path does not exist or is not writable we get into potential weird states still.